src/main/java/com/suse/saltstack/netapi/client/SaltStackClient.java

```diff
@@ -314,6 +315,33 @@ public Job call() throws SaltStackException {
     }
 
     /**
+     * Get previously run jobs.
+     * @return map containing run jobs keyed by job id.
+     * @throws SaltStackException if anything goes wrong
+     */
+    public Map<String, Job> getJobs() throws SaltStackException {
+         Result<List<Map<String, Job>>> result = connectionFactory
+                .create("/jobs", JsonParser.JOBS, config)
+                .getResult();
+         return result.getResult().get(0);
```

Comment:
@isbm it's not about being firmly guaranteed, it is about what is expected. We must make assumptions on what server returns based on its documented behavior - if it does not behave as it should, then it's OK for our code to behave in an undefined way, including failing with whatever `Exception` happens. We can't possibly foresee and wrap them all so it's better not to even attempt that and let the code fail "naturally" (examples: out of memory, hardware failures... will also generate cryptic stack traces, but they will be correct nonetheless).  Moreover `IndexOutOfRangeException` coming from here is pretty self-explanatory in this case.

If it is expected that sometimes there's no element, that's a whole different story of course.